### PR TITLE
Only load BlockMan in BlockMan member functions 

### PR DESCRIPTION
**Pull request by dongcarl**
The only important commit is "Only load BlockMan in BlockMan member functions", everything else is all just small style changes.

Here's the commit message, reproduced:
```
This commit effectively splits the "load block index itself" logic from
"derive Chainstate variables from loaded block index" logic.

This means that BlockManager::LoadBlockIndex{,DB} will only load what's
relevant to the BlockManager.
```

### CODE REVIEW

**Review by maflcko — APPROVED** [review]
LGTM

**Comment by ajtowns** [comment]
(Ditto what @MarcoFalke said)

File: src/node/blockstorage.cpp [hunk 1]
```diff
@@ -220,18 +220,21 @@ bool BlockManager::LoadBlockIndex(
...
+         [](const CBlockIndex* pa, const CBlockIndex* pb) {
+             return pa->nHeight < pb->nHeight;
+         });
 
     // Find start of assumed-valid region.
     int first_assumed_valid_height = std::numeric_limits<int>::max();
 
-    for (const auto& [height, block] : vSortedByHeight) {
-        if (block->IsAssumedValid()) {
+    for (CBlockIndex* pindex : vSortedByHeight) {
```
[the remark this answers is not in the document]
Comment (ryanofsky) [reply src/node/blockstorage.cpp]: In commit "style-only: No need for std::pair for vSortedByHeight" (6741a6957a240e9a0cdbd73df95f247b449e7b72)

This is fine but I don't really get it. Previous commit is renaming a `pindex` to `block_index`, this one is renaming a `block` variable to `pindex`. If these cleanups were consolidated into one commit, it'd be easier to see if the changes were consistent. (Also of the names I like simple `block` the best).

**Review by ryanofsky — APPROVED** [review]
Code review ACK af44e85b920055be4c3f724000a17f9d8d31d5ae. Ignore minor comments unless you feel like addressing them

File: src/node/blockstorage.cpp [hunk 2]
```diff
@@ -220,18 +220,21 @@ bool BlockManager::LoadBlockIndex(
...
     // Calculate nChainWork
-    std::vector<std::pair<int, CBlockIndex*>> vSortedByHeight;
+    std::vector<CBlockIndex*> vSortedByHeight;
     vSortedByHeight.reserve(m_block_index.size());
     for (auto& [_, block_index] : m_block_index) {
-        vSortedByHeight.push_back(std::make_pair(block_index.nHeight, &block_index));
+        vSortedByHeight.push_back(&block_index);
     }
-    sort(vSortedByHeight.begin(), vSortedByHeight.end());
+    sort(vSortedByHeight.begin(), vSortedByHeight.end(),
```
[the remark this answers is not in the document]
Comment (dongcarl) [reply src/node/blockstorage.cpp]: Done in e870d4740146a160aaea5e821437731d70345ff3

File: src/validation.cpp [hunk 3]
```diff
@@ -4083,8 +4083,80 @@ bool ChainstateManager::LoadBlockIndex()
...
-        bool ret = m_blockman.LoadBlockIndexDB(*this);
+        bool ret = m_blockman.LoadBlockIndexDB();
         if (!ret) return false;
+
+        std::vector<CBlockIndex*> vSortedByHeight;
+        vSortedByHeight.reserve(m_blockman.m_block_index.size());
+        for (auto& [_, block_index] : m_blockman.m_block_index) {
+            vSortedByHeight.push_back(&block_index);
+        }
+        sort(vSortedByHeight.begin(), vSortedByHeight.end(),
```
[the remark this answers is not in the document]
Comment (dongcarl) [reply src/validation.cpp]: Fixed in 28ba0313eac37e4a900b7e97af7169ce999c4024

File: src/node/blockstorage.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (dongcarl) [reply src/node/blockstorage.cpp]: Absorbed into 42e56d9b188f97c077ed2269e24acc0be35ece17

File: src/validation.cpp [hunk 4]
```diff
@@ -4083,8 +4083,80 @@ bool ChainstateManager::LoadBlockIndex()
...
+             [](const CBlockIndex* pa, const CBlockIndex* pb) {
+                 return pa->nHeight < pb->nHeight;
+             });
```
[the remark this answers is not in the document]
Comment (dongcarl) [reply src/validation.cpp]: See 28ba0313eac37e4a900b7e97af7169ce999c4024 and f865cf8ded2b2fbc82a6fbc41226d991909a6880

**Comment by dongcarl** [comment]
Pushed af44e85b920055be4c3f724000a17f9d8d31d5ae to f865cf8ded2b2fbc82a6fbc41226d991909a6880
- Addressed all outstanding minor comments

**Review by maflcko — APPROVED** [review]
review ACK f865cf8ded2b2fbc82a6fbc41226d991909a6880 🗂

<details><summary>Show signature</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

review ACK f865cf8ded2b2fbc82a6fbc41226d991909a6880 🗂
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUg0DAv/VZj7G6Rp0qriCmTSDHrk21I938vrPa39hBPwQ9GU3DpaKPoIth5mj2Vl
jMREh59ZzqKKorCPa0RdcVaRyiVPTRup0EOSi0b9qwIQREr1pumfJb1Mir9HJtCY
WbLA/sj1YGuXpIUvWAEUkKE56vpLVahWnSx5jd9QFO0DBWbfQ3iHWj/Soc5EBxEV
5pquK8ceeWyZxN7NKHWwKZegfjMmuPvmrr1HVfFB9qwjzYIsJk7Hv83PXbTHNzMa
OBBxlGy6o/anN1w+iS6tflGCzmTgxXXWKROlm2pCDAgOJZi4FyM2vj/V/9XM4E4F
RghIzqLltFmVJwvRV7x/o0Ahf3QHsez5TIl8JvFxVjBAwPP6WW2cbUQnXJRGrpho
oIUVnBf/r5iZCImOGzCkslm5oBDmWHCGi2jscPcdIQAcQNrusIN2jy4OrWxVAYFw
fjORBHiWF1TnQElKrfBurviwt+LO8SZFcjtTm7ZN4BGFFknCGgUBD9hqP9GB6mzt
Tu78aGDd
=BBgB
-----END PGP SIGNATURE-----
```


</details>

**Comment by ajtowns** [comment]
ACK f865cf8ded2b2fbc82a6fbc41226d991909a6880 ; code review only

**Pull request merged**